File: 345. Reverse Vowels of a String/Solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        read_vowels = ""
        for ch in s:
            if ch in "aeiouAEIOU":
                read_vowels = ch + read_vowels
        j=0
        for i in range(len(s)):
            if s[i] in "aeiouAEIOU":
                s = s[:i] + read_vowels[j] + s[i+1:]
                j+=1
        return s

File: 345. Reverse Vowels of a String/Solution.cpp
class Solution {
public:

    bool isVowel(char& c){
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
    }

    string reverseVowels(string s) {
        if (s.length() == 1){
            return s;
        }

        vector<char> read_vowels;
        int i=0;
        string output = "";

        for (char c: s){
            if (isVowel(c)){
                read_vowels.push_back(c);
                ++i;
            }
        }

        --i;

        for (int j = 0; j < s.length(); ++j){
            if (isVowel(s[j])){
                output += read_vowels[i--];
            } else{
                output += s[j];
            }
        }

        return output;
    }
};